app/cypress/support/listAuthenticatedPages.js

```diff
@@ -0,0 +1,31 @@
+const fs = require('fs').promises;
+const path = require('path');
+
+async function listAuthenticatedPages(
+  writeFile = 'cypress/support/authenticatedPages.json'
+) {
+  async function getRoutes(currentPath, ignoreFolders) {
+    const json = {};
+    const entries = await fs.readdir(currentPath, {withFileTypes: true});
+
+    const files = entries
+      .filter((file) => !file.isDirectory())
+      .map((file) => path.basename(file.name, '.tsx'));
+
+    const folders = entries.filter(
+      (folder) => folder.isDirectory() && !ignoreFolders.includes(folder.name)
+    );
+
+    for (const folder of folders) {
+      json[folder.name] = await getRoutes(`${currentPath}/${folder.name}`);
+    }
```

Comment:
If you want to keep async code and not ignore the linter warning, something like 
```js
const json = Promise
  .all(folders.map((folder) => getRoutes(`${currentPath}/${folder.name}`))
  .reduce((acc, curr, i) => acc[folders[i].name] = curr, {})
```
might work